src/test/java/org/broadinstitute/hellbender/utils/pileup/ReadPileupUnitTest.java

```diff
@@ -0,0 +1,345 @@
+package org.broadinstitute.hellbender.utils.pileup;
+
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static htsjdk.samtools.SAMRecord.NO_MAPPING_QUALITY;
+
+/**
+ * Test routines for read-backed pileup.
+ */
+public final class ReadPileupUnitTest {
```

Comment:
These `ReadPileup` tests are pretty inadequate -- most of the methods are barely tested (eg., one or two asserts that they return 0 or an empty collection). Recommend refactoring this class so that all public methods are covered by high-quality, meaningful test cases (would help if each method were tested separately instead of all together).